Ollivanders/src/net/pottercraft/Ollivanders2/OllivandersListener.java

```diff
@@ -662,128 +709,102 @@ public void onPlayerInteract (PlayerInteractEvent event)
       if (Ollivanders2.debug)
          p.getLogger().info("onPlayerInteract: enter");
 
-      if (action == null || player == null || event.getHand() != EquipmentSlot.HAND)
+      if (action == null || player == null)
       {
          return;
       }
 
       /**
-       * A left click is used to cast a spell
+       * A right or left click of the primary hand when holding a wand is used to make a magical action.
        */
-      if (action == Action.LEFT_CLICK_AIR || action == Action.LEFT_CLICK_BLOCK)
+      if ((event.getHand() == EquipmentSlot.HAND) && (p.holdsWand(player, EquipmentSlot.HAND)))
       {
-         if (Ollivanders2.debug)
-            p.getLogger().info("OllivandersListener:onPlayerInteract: left click action");
-
-         O2Player o2p = p.getO2Player(player);
-         Spells spell = o2p.getWandSpell();
-         if (spell != null)
+         /**
+          * A left click of the primary hand is used to cast a spell
+          */
+         if (action == Action.LEFT_CLICK_AIR || action == Action.LEFT_CLICK_BLOCK)
          {
-            double wandC;
-            boolean playerHoldsWand = p.holdsWand(player);
-            if (playerHoldsWand)
-            {
-               if (Ollivanders2.debug)
-                  p.getLogger().info("OllivandersListener:onPlayerInteract: player holds a wand");
-
-               wandC = p.wandCheck(player);
-               allyWand(player);
-            }
-            else
-            {
-               if (Ollivanders2.debug)
-               {
-                  p.getLogger().info("OllivandersListener:onPlayerInteract: player does not hold a wand");
-               }
+            if (Ollivanders2.debug)
+               p.getLogger().info("OllivandersListener:onPlayerInteract: left click action");
 
-               wandC = 10;
-            }
+            castSpell(player);
+            return;
+         }
+         /**
+          * A right click of the primary hand is used:
+          *  - to determine if the wand is the player's destined wand
+          *  - to set the mastered spell in the wand for non-verbal casting if they are sneaking while clicking
+          *  - to brew a potion if they are holding a glass bottle in their off hand and facing a cauldron
+          */
+         else if (action == Action.RIGHT_CLICK_AIR || action == Action.RIGHT_CLICK_BLOCK)
+         {
+            if (Ollivanders2.debug)
+               p.getLogger().info("OllivandersListener:onPlayerInteract: right click action");
 
-            createSpellProjectile(player, spell, wandC);
-            o2p.setSpellRecentCastTime(spell);
-            int spellc = p.getSpellNum(player, spell);
-            if (spellc < 100 || spell == Spells.AVADA_KEDAVRA || !playerHoldsWand)
+            Block cauldron = (playerFacingCauldron(player));
```

Comment:
Is this old or new potion stuff?

Reply:
I think this was in my previous commit and somehow got pulled in to this PR.